inc/admin.php

```diff
@@ -0,0 +1,149 @@
+ * loaded from localhost with an HTTPS protocol.
+ *
+ * @param string $tag    The HTML of the script tag to render.
+ * @param string $handle The registered script handle for this tag.
+ * @param string $src    The src URI of the JavaScript file this script loads.
+ * @return string The script tag HTML, conditionally transformed.
+ */
+function add_onerror_to_localhost_scripts( string $tag, string $handle, string $src ) : string {
+	if ( ! is_admin() || ! preg_match( get_localhost_pattern(), $src ) ) {
+		return $tag;
+	}
+	return preg_replace(
+		'/<script/',
+		'<script onerror="maybeSSLError && maybeSSLError( this );"',
+		$tag
+	);
+}
+
+/**
+ * Render inline JS into the page header to register a function which will be
+ * called should any of our registered HTTPS localhost scripts fail to load.
+ *
+ * If errors are detected when loading scripts from HTTPS localhost URLs, use
+ * WP's JavaScript notification system to display a contextual warning banner.
+ *
+ * @return void
+ */
+function render_localhost_error_detection_script() : void {
+	?>
+<script>
+( function() {
+	var scriptsWithErrors = [];
+
+	/**
+	 * @param HTMLScriptElement The script which experienced an error.
+	 */
+	window.maybeSSLError = function( script ) {
+		scriptsWithErrors.push( script );
+	};
+
+	/**
+	 * Check whether an error has occurred, then attempt to display a Block Editor
+	 * notice to alert the developer if so.
+	 *
+	 * @return void
+	 */
+	function processErrors() {
+		if ( ! scriptsWithErrors.length ) {
+			// There are no problems to highlight.
+			return;
+		}
+
+		var notices = null;
+		if ( window.wp && window.wp.data && window.wp.data.dispatch ) {
+			notices = window.wp.data.dispatch( 'core/notices' );
+		}
+		if ( ! notices ) {
+			// We're not in a context where it is easy to display a notice from JS.
+			return;
+		}
+
+		// Build a list of problem hosts.
+		var hosts = scriptsWithErrors.reduce(
+			function( hosts, script ) {
+				var src = script.getAttribute( 'src' );
+				if ( ! src || ! /https:\/\/localhost/i.test( src ) ) {
+					return hosts;
+				}
+				src = src.replace( /^(https:\/\/localhost:\d+).*$/i, '$1' );
+				hosts[ src ] = true;
+				return hosts;
+			},
+			{}
+		);
+		hosts = Object.keys( hosts );
+
+		// Build the error markup.
+		const messageHTML = [
+			'<strong>Error loading scripts from localhost!</strong>',
+			'<br>',
+			'Ensure that ',
+			( hosts.length > 1 ? 'these hosts are ' : 'this host is ' ),
+			'accessible, and that you have accepted any development server SSL certificates:',
+			'<ul>',
+			hosts.map( host => '<li><a target="_blank" href="' + host + '">' + host + '</a></li>' ).join( '' ),
+			'</ul>'
+		].join( '' );
+
+		notices.createErrorNotice( messageHTML, { __unstableHTML: true } );
+	}
+
+	// Set up processErrors to run 1 second after page load.
+	document.addEventListener( 'DOMContentLoaded', function() {
+		setTimeout( processErrors, 1000 );
+	} );
```

Comment:
I'm not opposed to adding this, but why is it needed? This code could only possibly run once, because we're rendering it as an action callback and repeatedly binding the action does not cause the callback to run more than once.

Reply:
That's corrrect. But the callback still takes up memory as it is still being registered (and as you say, it will never get executed again).